mesonbuild/wrap/wrap.py

```diff
@@ -587,6 +595,29 @@ def apply_patch(self) -> None:
                 raise WrapException(f'patch directory does not exist: {patch_dir}')
             self.copy_tree(src_dir, self.dirname)
 
+    def apply_diff_files(self) -> None:
+        for filename in self.wrap.diff_files:
+            mlog.log(f'Applying diff file "{filename}"')
+            path = Path(self.wrap.filesdir) / filename
+            if not path.exists():
+                raise WrapException(f'Diff file "{path}" does not exist')
+            if False and PATCH:
+                cmd = [PATCH, '-f', '-p1', '-i', str(path)]
+            elif GIT:
+                # If the `patch` command is not available, fall back to `git
+                # apply`. The `--work-tree` is necessary in case we're inside a
+                # Git repository: by default, Git will try to apply the patch to
+                # the repository root.
+                cmd = [GIT, '--work-tree', '.', 'apply', '-p1', str(path)]
```

Comment:
In the case it's a wrap-git, I'm wondering if we should use `git am` instead?

Reply:
As I said above, I believe it's better not to have it:
* This is surprising for the user: in this one case, the feature would do a different thing (not just apply patches to worktree, but also create commits); it also introduces an extra requirement for the patches to be in `git format-patch` format
* Special cases like this one are more likely to break (since they're less used).

If you disagree, I can add it back.

Reply:
I agree we should NOT use `git am`.

Reply:
ok